great_expectations/datasource/configured_pandas_data_asset.py

```diff
@@ -0,0 +1,214 @@
+import glob
+import logging
+import os
+from dataclasses import dataclass
+from typing import Dict, List, Optional
+
+from typing_extensions import Self
+
+from great_expectations.core.batch import Batch, BatchRequest
+from great_expectations.datasource.base_data_asset import BaseDataAsset
+from great_expectations.datasource.misc_types import (
+    BatchIdentifierException,
+    PassthroughParameters,
+    BatchIdentifiers,
+    NewBatchRequestBase,
+    NewConfiguredBatchRequest,
+)
+from great_expectations.datasource.data_connector.util import (
+    get_filesystem_one_level_directory_glob_path_list,
+)
+from great_expectations.validator.validator import Validator
+
+logger = logging.getLogger(__name__)
+
+class ConfiguredPandasDataAsset(BaseDataAsset):
+    def __init__(
+        self,
+        datasource,  # Should be of type: ConfiguredPandasDatasource,
+        name: str,
+        batch_identifiers: List[str] = ["filename"],
+        method: str = "read_csv",
+        base_directory: str = "",
+        regex: str = "(.*)",
+    ) -> None:
+
+        # !!! Check to make sure this is a valid configuration of parameters.
+        # !!! For example, if base_directory is Null, method and regex should be Null as well.
+        # !!! Also, if base_directory is not Null, method and regex should be populated as well. defaults are okay.
+        # !!! Also, if regex is not Null, the number of groups should be exactly equal to the number of parameters in batch_identifiers
+
+        self._method = method
+        self._base_directory = base_directory
+        self._regex = regex
+
+        super().__init__(
+            datasource=datasource, name=name, batch_identifiers=batch_identifiers
+        )
+
+    def update_configuration(
+        self,
+        name: Optional[str] = None,
+        batch_identifiers: Optional[List[str]] = None,
+        method: Optional[str] = None,
+        base_directory: Optional[str] = None,
+        regex: Optional[str] = None,
+    ) -> Self:
+
+        # !!! Check to make sure this is a valid configuration of parameters.
+        # !!! For example, if base_directory is Null, method and regex should be Null as well.
+        # !!! Also, if base_directory is not Null, method and regex should be populated as well. defaults are okay.
+        # !!! Also, if regex is not Null, the number of groups should be exactly equal to the number of parameters in batch_identifiers
+
+        if name != None:
+            self._datasource.rename_asset(self._name, name)
+
+        if batch_identifiers != None:
+            self._batch_identifiers = batch_identifiers
+
+        if method != None:
+            self._method = method
+
+        if base_directory != None:
+            self._base_directory = base_directory
+
+        if regex != None:
+            self._regex = regex
+
+        return self
+
+    def get_batch_request(
+        self, *batch_identifier_args, **batch_identifier_kwargs
```

Comment:
This is unnecessarily general I think. I also generally dislike kwargs from a usability stand-point since they are not self-documenting like named parameters or fields.

Reply:
Maybe.... In this case, I'm using args and kwargs to (1) avoid having to enumerate every argument that can be accepted by any pandas read_ method, and (2) to allow users to specify custom names for batch identifiers and pass them in as arguments.

I agree that it makes the docstring unhelpful.